tests/lib/rules/prefer-module-scope-constants.js

```diff
@@ -0,0 +1,64 @@
+const {RuleTester} = require('eslint');
+const rule = require('../../../lib/rules/prefer-module-scope-constants');
+
+const ruleTester = new RuleTester();
+
+const parserOptions = {ecmaVersion: 6, sourceType: 'module'};
+const moduleScopeErrors = [
+  {
+    kind: 'VariableDeclarator',
+    message:
+      'You must placee screaming snake case at module scope. If this is not meant to be a module-scoped variable, use camelcase instead.',
+  },
+];
+const nonConstErrors = [
+  {
+    kind: 'VariableDeclarator',
+    message:
+      'You must use `const` when using screaming snake case variables. If this is not a constant, use camelcase instead.',
+  },
+];
+
+ruleTester.run('prefer-module-scope-constants', rule, {
+  valid: [
+    {code: 'const FOO = true;', parserOptions},
+    {code: 'const foo = true;', parserOptions},
+    {code: '{ const foo = true; }', parserOptions},
+    {code: 'const foo = true, FOO = true;', parserOptions},
+    {code: 'const {FOO} = bar', parserOptions},
+    {code: '{ const {FOO} = bar; }', parserOptions},
+    {code: 'function foo() { const {FOO} = bar; }', parserOptions},
```

Comment:
Why does this pass?  Isn't `FOO` in `foo`'s scope?

Reply:
It's destructured, which I think we want to permit in case you did a namespace import from a module and want to get some of its properties.

Reply:
Makes sense 👍